provider_run_test.go

```diff
@@ -809,31 +811,37 @@ func TestPending(t *testing.T) {
 	t.Run("disallow_out_of_order", func(t *testing.T) {
 		ctx := context.Background()
 		fsys := newFsys()
-		p, err := goose.NewProvider(goose.DialectSQLite3, newDB(t), fsys,
-			goose.WithAllowOutofOrder(false),
-		)
-		check.NoError(t, err)
-		// Some migrations have been applied.
-		_, err = p.ApplyVersion(ctx, 1, true)
-		check.NoError(t, err)
-		_, err = p.ApplyVersion(ctx, 2, true)
-		check.NoError(t, err)
-		hasPending, err := p.HasPending(ctx)
-		check.NoError(t, err)
-		check.Bool(t, hasPending, true)
-		current, target, err := p.CheckPending(ctx)
-		check.NoError(t, err)
-		check.Number(t, current, 2)
-		check.Number(t, target, len(fsys))
-		_, err = p.Up(ctx)
-		check.NoError(t, err)
-		// All migrations have been applied.
-		hasPending, err = p.HasPending(ctx)
-		check.NoError(t, err)
-		check.Bool(t, hasPending, false)
-		current, target, err = p.CheckPending(ctx)
-		check.NoError(t, err)
-		check.Number(t, current, target)
+
+		run := func(t *testing.T, versionToApply int64) {
+			p, err := goose.NewProvider(goose.DialectSQLite3, newDB(t), fsys,
+				goose.WithAllowOutofOrder(false),
+			)
+			check.NoError(t, err)
+			// Some migrations have been applied.
+			_, err = p.ApplyVersion(ctx, 1, true)
+			check.NoError(t, err)
+			_, err = p.ApplyVersion(ctx, versionToApply, true)
+			check.NoError(t, err)
+			// TODO(mf): revisit the pending check behavior in addition to the HasPending
+			// method.
+			current, target, err := p.CheckPending(ctx)
```

Comment:
Given `CheckPending` only gets the current (db) and target (fsys) versions, I'm starting to reconsider the _naming_. 

Some alternatives

- VersionCheck
- PendingVersion
- CompareVersion
- GetVersions (there is a `GetDBVersion`) this might be a compliment

Reply:
Will tackle this in a separate PR, it's unreleased yet, so a breaking change here may be warranted. I think it'll just be a name change, leaning towards `GetVersions() (current, target int64, err error)`.